pkg/scheduler/framework/plugins/helper/normalize_score.go

```diff
@@ -17,33 +17,53 @@ limitations under the License.
 package helper
 
 import (
+	"math"
+
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 )
 
 // DefaultNormalizeScore generates a Normalize Score function that can normalize the
 // scores to [0, maxPriority]. If reverse is set to true, it reverses the scores by
 // subtracting it from maxPriority.
 func DefaultNormalizeScore(maxPriority int64, reverse bool, scores framework.NodeScoreList) *framework.Status {
-	var maxCount int64
+	var minCount int64 = math.MaxInt64
+	var maxCount int64 = math.MinInt64
 	for i := range scores {
 		if scores[i].Score > maxCount {
 			maxCount = scores[i].Score
 		}
+		if scores[i].Score < minCount {
+			minCount = scores[i].Score
+		}
 	}
 
-	if maxCount == 0 {
-		if reverse {
-			for i := range scores {
-				scores[i].Score = maxPriority
+	// This is necessary to preserve backwards-compatibility.
```

Comment:
I don't think this should be implicit. For example, you could have a scoring function that might have negative scores, but in the particular they are all positive, then you would have an unintended result.

I suggest that you create two exported functions as already proposed. To avoid repeated code, you can have the common bits in an unexported function.